src/api/analytics/utils.js

```diff
@@ -7,3 +7,21 @@ const whitelistRegExp = new RegExp(`(?:${whitelistURICodes.join('|')})`, 'g')
 
 export const customEncodeURIComponent = (uri) =>
     encodeURIComponent(uri).replace(whitelistRegExp, decodeURIComponent)
+
+export const formatRequestPath = ({ path, program, trackedEntityType }) =>
+    [path, program, trackedEntityType].filter(Boolean).join('/')
+
+export const formatDimension = ({
+    outputType,
+    programId,
+    programStageId,
+    dimension,
+}) =>
+    [
+        // XXX it would be clearer to have this consistent with what is sent in the request as for EVENT/ENROLLMENT
```

Comment:
For an uninitiated reader like myself, this comment is missing some context. From the text that is there, I gather that the current implementation is not ideal, and that you would have liked to have implemented things the same as EVENT/ENROLLMENT. I am guessing that the ternary operator below is an additional complexity that is preventing you doing so. But what is missing is a brief explanation of _why_ this additional complexity is present.

Reply:
This actually had to do with the `outputType` value.
Initially `TRACKED_ENTITY` was used, which didn't match with the value used in the api.
This has been updated tho, so this comment is not relevant anymore.